File: src/main/java/pw/react/backend/security/common/CommonUserDetailsService.java
package pw.react.backend.security.common;

import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import pw.react.backend.dao.UserRepository;
import pw.react.backend.models.UserEntity;

import java.util.Optional;

public class CommonUserDetailsService implements UserDetailsService {

    private final UserRepository userRepository;

    public CommonUserDetailsService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
        Optional<UserEntity> user = userRepository.findByEmail(email);
        if (user.isPresent()) {
            return user.get();
        } else {
            throw new UsernameNotFoundException("User not found with email: " + email);
        }
    }
}

File: src/main/kotlin/com/flatly/security/services/CommonUserDetailsService.kt
package com.flatly.security.services

import com.flatly.dao.UserRepository
import com.flatly.models.domain.toDomain
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException

class CommonUserDetailsService(private val userRepository: UserRepository) : UserDetailsService {

    override fun loadUserByUsername(username: String): UserDetails {
        val user = userRepository.findByUsername(username)?.toDomain()
        return user ?: throw UsernameNotFoundException("User not found with username: $username")
    }
}
